utils/wasm-gen/src/generator/syscalls/invocator.rs

```diff
@@ -794,3 +762,42 @@ impl From<DisabledSyscallsInvocator> for ModuleWithCallIndexes {
         }
     }
 }
+
+#[derive(Clone, Debug)]
+struct ParamInstructions(Vec<Instruction>);
```

Comment:
```suggestion
struct ParamSetters(Vec<Instruction>);
```
and also it'd be nice to have doc-comment on it as it's not very clear what's the purpose of the inner `Vec`

Reply:
Anyway I think that squashing `ParamSetter` and `SyscallInstructions` is not a very good idea as the interface of newly introduced struct makes to think that you're working with a single value when it actually contains multiple

Reply:
I don't think `ParamSetters` actually reflects purposes of the type and it's methods. There are no `set`s or something similar. What's more, when some type is called as `ParamInstructions`, doesn't name tell a reader that it is some wrapper for instructions which define syscall param? It becomes obvious when you see that you wrap `Vec<Instruction>`.

In a word, I'm against renaming it to `ParamSetters`. That's more vague name in the context of the type usage.

Reply:
> Anyway I think that squashing `ParamSetter` and `SyscallInstructions` is not a very good idea as the interface of newly introduced struct makes to think that you're working with a single value when it actually contains multiple

Sorry, I didn't catch. You actually have a separate instance for every syscall param/value.

Reply:
discussed